lib/common/forms-rule-engine.js

```diff
@@ -1396,9 +1593,21 @@
       return cb(new Error("Should not submit section field: " + fieldDefinition.name));
     }
 
-    function rulesResult(rules, cb) {
+    /**
+     *
+     * Processing all of the rules that target a single field to check if the rule is active or not.
+     *
+     * @param {string|null} fieldId - The ID of the field being checked. If it is null then it is a page rule being checked.
+     * @param {Array} rules
+     * @param {number} sectionIndex
+     * @param {function} cb
+     */
+    function rulesResult(fieldId, rules, sectionIndex, cb) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Function has a complexity of 23 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)